master/internal/trial.go

```diff
@@ -160,80 +159,86 @@ type terminatedContainerWithState struct {
-
-	warmStartCheckpointID *int
-
-	create searcher.Create
-	close  *searcher.Close
-
-	sequencer *trialWorkloadSequencer
-
-	// restarts is essentially a failure count, it increments when the trial fails and we retry it.
-	restarts int
-
-	// runID is a count of how many times the task container(s) have stopped and restarted, which
-	// could be due to a failure or due to normal pausing and continuing. When runID increments,
-	// it effectively invalidates any outstanding terminateTimeout messages so that we don't
-	// accidentally kill a fresh container due to the terminateTimeout message from an older
-	// container.
-	runID int
-
-	replaying bool
-
-	// The following fields tracks the reasons for termination.
-	earlyExit                  bool
-	pendingGracefulTermination bool
-	terminationSent            bool
-	cancelUnready              bool
-	killed                     bool
-
-	// The following fields tracks the interaction with the resource providers.
-	task        *resourcemanagers.AllocateRequest
-	allocations []resourcemanagers.Allocation
-
-	// The following fields tracks containers and their states.
-	lastContainerConnectedTime time.Time
-	startedContainers          map[cproto.ID]bool
-	containers                 map[cproto.ID]cproto.Container // only for running containers.
-	containerRanks             map[cproto.ID]int              // only for launched containers.
-	containerAddresses         map[cproto.ID][]cproto.Address // only for running containers.
-	containerSockets           map[cproto.ID]*actor.Ref       // only for running containers.
-	terminatedContainers       map[cproto.ID]terminatedContainerWithState
-	// tracks if allReady check has passed successfully.
-	allReadySucceeded bool
-
-	agentUserGroup *model.AgentUserGroup
-	taskSpec       *tasks.TaskSpec
-	privateKey     []byte
-	publicKey      []byte
-}
+type (
+	// trialState keeps all the state of the trial that we persist between failures
+	trialState struct {
+		TrialWorkloadSequencerState json.RawMessage
+
+		// restarts is essentially a failure count, it increments when the trial fails and we retry it.
+		Restarts int
+
+		// RunID is a count of how many times the task container(s) have stopped and restarted, which
+		// could be due to a failure or due to normal pausing and continuing. When RunID increments,
+		// it effectively invalidates any outstanding terminateTimeout messages so that we don't
+		// accidentally kill a fresh container due to the terminateTimeout message from an older
+		// container.
+		RunID int
+
+		// The following fields tracks the reasons for termination.
+		EarlyExit                  bool
+		PendingGracefulTermination bool
+		TerminationSent            bool
+		CancelUnready              bool
+		Killed                     bool
+	}
+
+	// trial is an actor which is responsible for handling:
+	//  - messages from the scheduler,
+	//  - messages from the experiment,
+	//  - messages from the trial container(s),
+	//  - replay logic, and
```

Comment:
several comments need updating in this file; this is one of them.  I'm totally fine if you leave that for the last step though.